Security/Authentication/Token/OAuth2TokenFactory.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Trikoder\Bundle\OAuth2Bundle\Security\Authentication\Token;
+
+use Psr\Http\Message\ServerRequestInterface;
+use Symfony\Component\Security\Core\User\UserInterface;
+
+class OAuth2TokenFactory
+{
+    /**
+     * @var string
+     */
+    private $rolePrefix;
+
+    /**
+     * OAuth2TokenFactory constructor.
+     */
+    public function __construct(string $rolePrefix)
```

Comment:
@rjwebdev I noticed that you made the role prefix on the `OAuth2Token` value object nullable, but here it is not. We need to be consistent.

A follow-up question: Do we want to allow the users to omit the prefix entirely?

Depending on which route we decide to take we'll have to add either `cannotBeEmpty` or `treatNullLike` options to the configuration node.

Reply:
Didn't remove the optional parameter in the `OAuth2Token` constructor

Not sure what you mean with your follow-up question